src/lazy/bytes_ref.rs

```diff
@@ -1,25 +1,24 @@
 use crate::text::text_formatter::IonValueFormatter;
 use crate::Bytes;
-use std::borrow::Cow;
 use std::fmt::{Debug, Display, Formatter};
 use std::ops::Deref;
 
-#[derive(Clone)]
+#[derive(Copy, Clone)]
 pub struct BytesRef<'data> {
-    data: Cow<'data, [u8]>,
+    data: &'data [u8],
```

Comment:
🗺️ Having types like this implement `Copy` (because decoding can use the bump allocator instead of the heap) should offer a nice speed boost. It also makes a lot of the APIs easier to work with -- every 'lazy' item type is now also `Copy`.